File: jna/src/main/java/dark/image.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

public class image extends Structure implements Structure.ByValue {
    public int w;
    public int h;
    public int c;
    public Pointer data;

    public image() {
        super();
    }

    public image(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("w","h","c","data");
    }
}

File: jna/src/main/java/dark_kotlin/image.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure

open class image: Structure, Structure.ByValue {

    @JvmField var w = 0
    @JvmField var h = 0
    @JvmField var c = 0
    @JvmField var data: Pointer? = null //float*

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("w", "h", "c", "data")
    }
}
